payments-core/src/main/java/com/stripe/android/payments/core/authentication/RedirectResolver.kt

```diff
@@ -0,0 +1,44 @@
+package com.stripe.android.payments.core.authentication
+
+import java.net.HttpURLConnection
+import java.net.URL
+import javax.inject.Inject
+import javax.net.ssl.HttpsURLConnection
+
+private const val RedirectTimeoutInMillis = 10_000
+
+internal fun interface RedirectResolver {
+    suspend operator fun invoke(url: String): String
+}
+
+internal typealias ConfigureSslHandler = HttpsURLConnection.() -> Unit
+
+internal class RealRedirectResolver(
+    private val configureSSL: ConfigureSslHandler,
+) : RedirectResolver {
+
+    @Inject
+    constructor() : this(configureSSL = {})
+
+    override suspend fun invoke(url: String): String {
+        return runCatching {
+            val connection = (URL(url).openConnection() as HttpURLConnection).apply {
+                connectTimeout = RedirectTimeoutInMillis
```

Comment:
another option would be to use a whole timeout (like wrap the whole thing in a coroutines withTimeout).

That _might_ be more along the lines of what we want.